ruff_cli/src/printer.rs

```diff
@@ -336,6 +336,50 @@ impl<'a> Printer<'a> {
         Ok(())
     }
 
+    pub fn write_statistics(&self, diagnostics: &Diagnostics) -> Result<()> {
+        let mut violations = diagnostics
+            .messages
+            .iter()
+            .map(|message| message.kind.rule())
+            .collect::<Vec<_>>();
+        violations.sort();
+        violations.dedup();
+
+        let mut stdout = BufWriter::new(io::stdout().lock());
+        match self.format {
+            SerializationFormat::Json => {}
+            SerializationFormat::Junit => {}
+            SerializationFormat::Text => {
+                for rule in violations {
+                    let count = diagnostics
+                        .messages
+                        .iter()
+                        .filter(|message| message.kind.rule() == rule)
+                        .count();
+                    let message = diagnostics
+                        .messages
+                        .iter()
+                        .filter(|message| message.kind.rule() == rule)
+                        .map(|message| message.kind.body())
```

Comment:
can you help me here? I don't want to map every rule but only the `.first()` one but don't know how to get the first instance. `.first()` gives me only errors.

Reply:
I think you want something like:

```rust
let message: Option<String> = diagnostics.messages.iter().find(|message| message.kind.rule() == rule).map(|message| message.kind.body());
if let Some(message) = message{
  writeln!(...)
}
```

Reply:
Note that the `.map` in my version and your version are slightly different. In your version, you're `.map`-ing over an iterator, kind of like Python's `map()`. In my version, it's `map` on an `Option<String>`, which basically means: if the value is `None`, return `None`; if the value is `Some(String)`, run this function over `String`.

Reply:
i try to further refactor, so that we can generate also JSON output.

Reply:
@charliermarsh enhanced in 0d66a24bacbaec144351e5a9aba36483613cdc03 - what do you think?

The JSON output is:
```
[
  {
    "count": 3,
    "code": "E101",
    "message": "Indentation contains mixed spaces and tabs"
  },
]
```
would you prefer?:
```
{
  "E101": {                                                                                                                                                                         
      "count": 3,                                                     
      "message": "Indentation contains mixed spaces and tabs"       
   }, 
}
```

and the output is just appended to the regular output. would one want to combine it in case of the JSON formatter? i am unsure how to handle this. there would be two different items in the list then.